File: frontend/board/src/app/modules/kanban/models/sprint.ts
import { Quarter } from "./quarter"
import { Team } from "./team"

export enum SprintStatus {
  RUNNING = 'running',
  COMPLETED = 'completed',
  WAITING = 'waiting'
}

export type Sprint = {
  id: number
  title: string
  represent: string
  start_date: string
  end_date: string
  team_id: number
  team: Team
  status: SprintStatus
  quarter: Quarter
  hours_per_user: number | null
  can_run: boolean
  can_delete: boolean
  can_complete: boolean
  count_bindings: number
}

export type SaveSprintRequest = {
  id?: number
  title: string
  start_date: string
  end_date: string
  team_id: number
  hours_per_user: number | null
}

File: backend/internal/infra/persistance/models/sprint.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Sprint struct {
	gorm.Model
	Id           uint      `gorm:"id;primaryKey"`
	Status       string    `gorm:"status;default:waiting;not null"`
	TeamId       uint      `gorm:"team_id;not null"`
	HoursPerUser uint      `gorm:"hours_per_user;default:0;not null"`
	Title        string    `gorm:"title;default:''"`
	StartDate    time.Time `gorm:"start_date;not null"`
	EndDate      time.Time `gorm:"end_date;not null"`
	Team         Team      `gorm:"foreignKey:team_id;not null"`
}
